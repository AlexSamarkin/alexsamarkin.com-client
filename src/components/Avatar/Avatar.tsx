import React from 'react';

export interface AvatarProps {
    avatar: string;
}

export const Avatar: React.FC<AvatarProps> = ({ avatar }) => {
    return (
        <svg className="avatar avatar--180" viewBox="0 0 188 188">
            <g className="avatar__box">
                <image xlinkHref={avatar} height="100%" width="100%" />
            </g>
        </svg>
    );
};

export default Avatar;
