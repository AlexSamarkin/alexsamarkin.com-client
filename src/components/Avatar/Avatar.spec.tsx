import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Avatar from './Avatar';

Enzyme.configure({ adapter: new Adapter() });

describe('Avatar component', () => {
    test('renders', () => {
        const wrapper = shallow(<Avatar avatar="no-image.png" />);
        expect(wrapper.exists()).toBe(true);
        expect(wrapper).toMatchSnapshot();
    });
});
