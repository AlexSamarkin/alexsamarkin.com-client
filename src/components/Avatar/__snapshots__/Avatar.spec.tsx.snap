// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar component renders 1`] = `ShallowWrapper {}`;
